### ISSUE TITLE
Job status indefinitely RUNNING but rsnapshot is not running

### ISSUE THREAD

**Issue by xezpeleta** [elkarbackup/elkarbackup#82]
One of my backup job status is still RUNNING after more than 12 hours. There is no rsnapshot/rsync process running and pid file is missing. As a result, next jobs are still QUEUED. There is not a error message in the log.

Apperently Rsnapshot died suddenly and Elkarbackup didn't realize. It only happened once, but I think it's important.

Elkarbackup v1.1.3 (stable)
Debian 7 Wheezy

**Comment by xezpeleta** [comment]
Related issue #40

**Comment by xezpeleta** [comment]
The problem has been solved running the troubled job manually (`Run now` button): this time the job finished successfully but next jobs were still QUEUED. Next day, after the next backup, everything was OK.

I can't reproduce this issue again, so if there is no news in some days I'll close it.

**Comment by xezpeleta** [comment]
Same problem again. This time I have more information.

![screenshot from 2016-02-04 12-16-16](https://cloud.githubusercontent.com/assets/1846038/12813679/db0674fe-cb39-11e5-816f-571dacf43604.png)

Scenario:
* File `/tmp/rsnapshot.14_33.cfg` exists with a correct rsnapshot config
* PID file not present under `/var/run` directory
* Rsnapshot or rsync not running

LOGS (**from web interface**, MySQL database):

![screenshot from 2016-02-04 12-16-39](https://cloud.githubusercontent.com/assets/1846038/12813787/9d7575a8-cb3a-11e5-91e1-65556813a3bc.png)

...but....

LOGS (**from logfile** `/var/log/elkarbackup/BnvLog.log`):

```
...
[2016-02-03 22:49:07] BnvWeb.INFO: RUNNING {"source":"StatusReport","link":"/client/14/job/31"}  
[2016-02-03 22:49:07] BnvWeb.INFO: Running "/usr/bin/rsnapshot" -c "/tmp/rsnapshot.14_31.cfg" sync 2>&1 {"source":"TickCommand","link":"/client/14/job/31"}  
[2016-02-03 22:49:17] BnvWeb.INFO: Command "/usr/bin/rsnapshot" -c "/tmp/rsnapshot.14_31.cfg" sync 2>&1 succeeded with output:  {"source":"TickCommand","link":"/client/14/job/31"}  
[2016-02-03 22:49:17] BnvWeb.INFO: Running "/usr/bin/rsnapshot" -c "/tmp/rsnapshot.14_31.cfg" Daily 2>&1 {"source":"TickCommand","link":"/client/14/job/31"}  
[2016-02-03 22:49:18] BnvWeb.INFO: Command "/usr/bin/rsnapshot" -c "/tmp/rsnapshot.14_31.cfg" Daily 2>&1 succeeded with output:  {"source":"TickCommand","link":"/client/14/job/31"}  
[2016-02-03 22:49:18] BnvWeb.INFO: Client "14", Job "31" du begin. {"source":"TickCommand","link":"/client/14/job/31"}  
[2016-02-03 22:49:19] BnvWeb.INFO: Client "14", Job "31" du end. {"source":"TickCommand","link":"/client/14/job/31"}  
[2016-02-03 22:49:19] BnvWeb.INFO: Client "14", Job "31" ok. {"source":"TickCommand","link":"/client/14/job/31"}  
[2016-02-03 22:49:19] BnvWeb.INFO: OK {"source":"StatusReport","link":"/client/14/job/31"}  
[2016-02-03 22:49:19] BnvWeb.INFO: RUNNING {"source":"StatusReport","link":"/client/14/job/32"}  
[2016-02-03 22:49:19] BnvWeb.INFO: Running "/usr/bin/rsnapshot" -c "/tmp/rsnapshot.14_32.cfg" sync 2>&1 {"source":"TickCommand","link":"/client/14/job/32"}  
[2016-02-03 22:49:31] BnvWeb.INFO: Command "/usr/bin/rsnapshot" -c "/tmp/rsnapshot.14_32.cfg" sync 2>&1 succeeded with output:  {"source":"TickCommand","link":"/client/14/job/32"}  
[2016-02-03 22:49:31] BnvWeb.INFO: Running "/usr/bin/rsnapshot" -c "/tmp/rsnapshot.14_32.cfg" Daily 2>&1 {"source":"TickCommand","link":"/client/14/job/32"}  
[2016-02-03 22:49:31] BnvWeb.INFO: Command "/usr/bin/rsnapshot" -c "/tmp/rsnapshot.14_32.cfg" Daily 2>&1 succeeded with output:  {"source":"TickCommand","link":"/client/14/job/32"}  
[2016-02-03 22:49:31] BnvWeb.INFO: Client "14", Job "32" du begin. {"source":"TickCommand","link":"/client/14/job/32"}  
[2016-02-03 22:49:31] BnvWeb.INFO: Client "14", Job "32" du end. {"source":"TickCommand","link":"/client/14/job/32"}  
[2016-02-03 22:49:31] BnvWeb.INFO: Client "14", Job "32" ok. {"source":"TickCommand","link":"/client/14/job/32"}  
[2016-02-03 22:49:31] BnvWeb.INFO: OK {"source":"StatusReport","link":"/client/14/job/32"}  
[2016-02-03 22:49:31] BnvWeb.INFO: RUNNING {"source":"StatusReport","link":"/client/14/job/33"}  
[2016-02-03 22:49:31] BnvWeb.INFO: Running "/usr/bin/rsnapshot" -c "/tmp/rsnapshot.14_33.cfg" sync 2>&1 {"source":"TickCommand","link":"/client/14/job/33"}  
[2016-02-03 22:50:01] BnvWeb.INFO: Nothing to run. {"source":"TickCommand"}  
[2016-02-03 22:51:01] BnvWeb.INFO: Nothing to run. {"source":"TickCommand"}  
[2016-02-03 22:52:01] BnvWeb.INFO: Nothing to run. {"source":"TickCommand"}  
[2016-02-03 22:53:02] BnvWeb.INFO: Nothing to run. {"source":"TickCommand"}  
[2016-02-03 22:54:02] BnvWeb.INFO: Nothing to run. {"source":"TickCommand"}  
...
...
[2016-02-04 08:00:01] BnvWeb.INFO: Nothing to run. {"source":"TickCommand"}  
[2016-02-04 08:01:01] BnvWeb.INFO: Nothing to run. {"source":"TickCommand"}  
[2016-02-04 08:02:01] BnvWeb.INFO: Nothing to run. {"source":"TickCommand"}  
[2016-02-04 08:03:02] BnvWeb.INFO: Nothing to run. {"source":"TickCommand"}  
[2016-02-04 08:03:45] BnvWeb.INFO: Command "/usr/bin/rsnapshot" -c "/tmp/rsnapshot.14_33.cfg" sync 2>&1 succeeded with output:  {"source":"TickCommand","link":"/client/14/job/33"}  
[2016-02-04 08:03:45] BnvWeb.INFO: Running "/usr/bin/rsnapshot" -c "/tmp/rsnapshot.14_33.cfg" Daily 2>&1 {"source":"TickCommand","link":"/client/14/job/33"}  
[2016-02-04 08:04:02] BnvWeb.INFO: Nothing to run. {"source":"TickCommand"}  
[2016-02-04 08:05:01] BnvWeb.INFO: Nothing to run. {"source":"TickCommand"}  
[2016-02-04 08:06:02] BnvWeb.INFO: Nothing to run. {"source":"TickCommand"}  
[2016-02-04 08:07:01] BnvWeb.INFO: Nothing to run. {"source":"TickCommand"}  
[2016-02-04 08:08:01] BnvWeb.INFO: Nothing to run. {"source":"TickCommand"}  
[2016-02-04 08:08:50] BnvWeb.INFO: Command "/usr/bin/rsnapshot" -c "/tmp/rsnapshot.14_33.cfg" Daily 2>&1 succeeded with output:  {"source":"TickCommand","link":"/client/14/job/33"}  
[2016-02-04 08:08:50] BnvWeb.INFO: Client "14", Job "33" du begin. {"source":"TickCommand","link":"/client/14/job/33"}  
[2016-02-04 08:09:02] BnvWeb.INFO: Nothing to run. {"source":"TickCommand"}  
[2016-02-04 08:10:01] BnvWeb.INFO: Nothing to run. {"source":"TickCommand"}  
[2016-02-04 08:11:02] BnvWeb.INFO: Nothing to run. {"source":"TickCommand"}  
[2016-02-04 08:11:36] BnvWeb.INFO: Client "14", Job "33" du end. {"source":"TickCommand","link":"/client/14/job/33"}  
[2016-02-04 08:12:01] BnvWeb.INFO: Nothing to run. {"source":"TickCommand"}  
[2016-02-04 08:13:01] BnvWeb.INFO: Nothing to run. {"source":"TickCommand"}  
[2016-02-04 08:14:01] BnvWeb.INFO: Nothing to run. {"source":"TickCommand"}  
[2016-02-04 08:15:01] BnvWeb.INFO: Nothing to run. {"source":"TickCommand"}
```

So the job finished successfully. Apparently, it failed writing logs to the database. Moreover, elkarbackup didn't continue with the following QUEUED jobs after finishing the job 33.

Any help would be appreciated: @antton @elacunza @blaskurain 

Thanks!

**Comment by xezpeleta** [comment]
We saw this bug several times.

Unfortunatelly I can't reproduce this bug again in my production server. After running the job manually (with "run now" button), everything works well again. Maybe I should try to backup the same data (more than 500GB) in a clean test environment to reproduce the problem.

Not sure if it makes sense, but in my opinion this strange bug could be caused by `du`  command:

https://github.com/elkarbackup/elkarbackup/blob/master/src/Binovo/ElkarBackupBundle/Lib/BackupRunningCommand.php#L222

That's the only command executed during the runJob function without error control.

**Comment by xezpeleta** [comment]
This bug should be fixed in the following version v1.2.0.

We reverted a change done in a previous version that may provoke this error in some circumstances. Now the command "du" is executed again after finishing the job backup.
